### DISCUSSION TITLE
does subscribeWithSelector need an unsubscribe method?

### DISCUSSION THREAD

**Discussion by Need-an-AwP** [pmndrs/zustand#2910] [Q&A] [answered]
im using the subscribeWithSelector middleware to subscribe some data from my store, like useeffect in react (im not very sure if this is the proper way of using store)
dose the subscribe have any unsubscribe method to clean it when the component is uninstall?
or i just want it stop call the callback func, what should i do
i havent found any info about how to clean the subscriber in the doc

**Comment by dbritto-dev** [comment, marked as answer]
@Need-an-AwP the `subscribe` method returns the `unsubscribe` method -> https://zustand.docs.pmnd.rs/apis/create#subscribing-to-state-updates

**Reply by Need-an-AwP** [reply]
thanks
i just found subscribeWithSelector middleware seems like doesnt have an unsubscribe method
but it's ok to use the normal subscribe method

**Reply by dbritto-dev** [reply]
@Need-an-AwP should have an unsubscribe method, that subscribe with selector just extend the subscribe functionality
